libqaul/common/src/payload.rs

```diff
@@ -0,0 +1,68 @@
+//! Types related to data being moved around and its verification.
+use crate::HashBytes;
+use blake2::{Blake2b, Digest};
+use generic_array::GenericArray;
+
+/// The actual content of a message, along with the mechanism to validate that
+/// no transmission errors occurred (message digest of payload type and data).
+#[derive(PartialEq, Eq, Debug)]
+pub struct Payload {
+    length: u64,
+    data: Vec<u8>,
+    digest: HashBytes
+}
+
+// Compute and return the 64-bit Blake2b digest of the given data.
+fn blake2b_digest(data: &[u8]) -> HashBytes { Blake2b::digest(data) }
```

Comment:
Code-style wise we don't currently enforce the rule of `rustfmt` (we will at some point probably) but I'm kinda against "in-lining" function bodies like this. 

I've read too much C code and my brain just parses it as a header definition and then I get confused about what it is in Rust :sweat_smile:

Reply:
OK! I can just rustfmt everything prior committing, that might be best